Makefile

```diff
@@ -142,33 +144,31 @@ export LDFLAGS :=       \
 # * Kernel Build Targets *
 # ************************
 
-all: debug release
-
-# Debug build
-debug: CPPFLAGS += -DDEBUG
-debug: CXXFLAGS += -ggdb3
-debug: CFLAGS += -ggdb3
-debug: $(KERNEL)
-
-# Release build
-# This will be build by default since it
-# is the first target in the Makefile
-release: CXXFLAGS += -O3 -mno-avx
-release: CFLAGS += -O3 -mno-avx
-release: $(KERNEL)
+ifeq ($(MODE), "debug")
+    CPPFLAGS += -DDEBUG
+	CXXFLAGS += -ggdb3
+	CFLAGS += -ggdb3
+else
+    CPPFLAGS += -DRELEASE
+	CXXFLAGS += -O3 -mno-avx
+	CFLAGS += -O3 -mno-avx
+endif
 
 # Kernel (Linked With Libraries)
-.PHONY: $(KERNEL)
-$(KERNEL):
-	@printf "$(COLOR_INFO)Making Libs$(COLOR_NONE)\n"
+$(PRODUCTS_DIR)/$(MODE)/$(KERNEL):
+	@printf "$(COLOR_INFO)Making Libs ($(MODE))$(COLOR_NONE)\n"
 	@for dir in $(LIB_DIRS); do        \
         $(MAKE) -C $$dir $(PROJ_NAME); \
     done
-	@printf "$(COLOR_INFO)Making Kernel$(COLOR_NONE)\n"
+	@printf "$(COLOR_INFO)Making Kernel ($(MODE))$(COLOR_NONE)\n"
 	@$(MAKE) -C $(KERNEL) $(KERNEL)
 	@printf "$(COLOR_INFO)Done!$(COLOR_NONE)\n"
 
-$(PRODUCTS_DIR)/$(KERNEL): $(KERNEL)
+# Hacky way to build all targets. This target cannot
+# be the first one otherwise it's a circular dependency.
+all:
+	@$(MAKE) MODE=debug
+	@$(MAKE) MODE=release
```

Comment:
Lol this is how I assumed you would have solved this, but it still looks funny (no changes required)